src/_P039_Thermocouple.ino

```diff
@@ -139,28 +302,104 @@ boolean Plugin_039(byte function, struct EventStruct *event, String& string)
+        addFormSelector(F("Sensor Family Type"), F("P039_famtype"), 2, Foptions, FoptionValues, family);
+        addFormNote(F("Set sensor family of connected sensor - thermocouple or RTD. Submit the form after choice to allow update of sections below accordingly !"));
+      }
 
+      addFormSubHeader(F("Device Selection"));
+      
       const byte choice = P039_MAX_TYPE;
-      {
-        const String options[3]      = {   F("MAX 6675"), F("MAX 31855"), F("MAX 31856") };
-        const int    optionValues[3] = { P039_MAX_6675, P039_MAX_31855, P039_MAX_31856 };
-        addFormSelector(F("Adapter IC"), F("p039_maxtype"), 3, options, optionValues, choice);
+
+      if (family == P039_TC){
+
+        {
+          const String options[3]      = {   F("MAX 6675"), F("MAX 31855"), F("MAX 31856") };
+          const int    optionValues[3] = { P039_MAX_6675, P039_MAX_31855, P039_MAX_31856 };
+          addFormSelector(F("Adapter IC"), F("P039_maxtype"), 3, options, optionValues, choice);
+          addFormNote(F("Set adapter IC used. Submit the form after choice to allow update of sections below accordingly !"));
+        }
+
+        addFormSubHeader(F("Sub-Device Settings"));
+
+        if (choice == P039_MAX_31856) {
+          addFormNote(F("Set Thermocouple type for MAX31856"));
+          const String Toptions[8]      = { F("B"), F("E"), F("J"), F("K"), F("N"), F("R"), F("S"), F("T") };
+          const int    ToptionValues[8] = { 0, 1, 2, 3, 4, 5, 6, 7 };
+          addFormSelector(F("Thermocouple type"), F("P039_tctype"), 8, Toptions, ToptionValues, P039_TC_TYPE);
+        }
       }
+      else {
+        
+        {
+          const String TPoptions[2] = {F("MAX 31865"), F("LM7x")};
+          const int TPoptionValues[2] = {P039_MAX31865, P039_LM7x};
+          addFormSelector(F("Adapter IC"), F("P039_maxtype"), 2, TPoptions, TPoptionValues, choice);
+          addFormNote(F("Set used RTD Converter Module. Currently only MAX31865 is fully supported. LM7x derivatives are untested and experimental. Submit the form after choice to allow update of sections below accordingly !"));
+        }
 
-      if (choice == P039_MAX_31856) {
-        addFormNote(F("Set Thermocouple type for MAX31856"));
-        const String Toptions[8]      = { F("B"), F("E"), F("J"), F("K"), F("N"), F("R"), F("S"), F("T") };
-        const int    ToptionValues[8] = { 0, 1, 2, 3, 4, 5, 6, 7 };
-        addFormSelector(F("Thermocouple type"), F("p039_tctype"), 8, Toptions, ToptionValues, P039_TC_TYPE);
+        addFormSubHeader(F("Sub-Device Settings"));
+
+        if (choice == P039_MAX31865)
+        {
+          const String PToptions[2] = {F("PT100"), F("PT1000")};
+          const int PToptionValues[2] = {MAX31865_PT100, MAX31865_PT1000};
+          addFormSelector(F("Resistor Type"), F("P039_rtdtype"), 2, PToptions, PToptionValues, P039_RTD_TYPE);
+          addFormNote(F("Set Resistor Type for MAX31865"));
```

Comment:
Please place these per form selector (and array of strings + options) in a separate scope (`{ ... }`) so they are destructed as soon as they are no longer needed.
With arrays of 2 elements it will probably not make a lot of difference in memory usage, but it is done a good practice to keep memory usage as low as possible.

Reply:
OK,

now this explains the {...} groups in the source code ! I think i get the idea - i will update the code.

Reply:
Code updated - changes available for review.